src/main/java/hudson/remoting/Channel.java

```diff
@@ -171,7 +172,7 @@
      * Records the {@link Request}s being executed on this channel, sent by the remote peer.
      */
     /*package*/ final Map<Integer,Request<?,?>> executingCalls =
-        Collections.synchronizedMap(new Hashtable<>());
+        Collections.synchronizedMap(new HashMap<>());
```

Comment:
Similar comment here. Why not ConcurrentHashMap?

And the thread safety appears broken both before and after.

(If we're going to change these, we probably ought to make sure the thread safety is solid, otherwise it's probably better not to change.)

Reply:
In this particular case, there's no reason to wrap Hashtable with synchronizedMap as that just layers two equivalent locks on the same collection.

Reply:
I'm not understanding what you mean? If it doesn't need to be wrapped with synchronizedMap, why does your change propose to still do that?

Reply:
Because you should only use one or the other.